### PR TITLE
[Merged by Bors] - chore: merge bump/v4.12.0 and move toolchain to v4.12.0-rc1

### PR DESCRIPTION
**Pull request by kim-em**

### CODE REVIEW

**Comment by semorrison** [comment]
bors p=10

File: Mathlib/Data/List/Infix.lean [hunk 1]
```diff
@@ -163,93 +83,28 @@ theorem concat_get_prefix {x y : List α} (h : x <+: y) (hl : x.length < y.lengt
...
--- Alternatively, use mem_tails
-instance decidableSuffix [DecidableEq α] : ∀ l₁ l₂ : List α, Decidable (l₁ <:+ l₂)
-  | [], l₂ => isTrue ⟨l₂, append_nil _⟩
-  | a :: l₁, [] => isFalse <| mt (Sublist.length_le ∘ IsSuffix.sublist) (by simp)
-  | l₁, b :: l₂ =>
-    @decidable_of_decidable_of_iff _ _
-      (@instDecidableOr _ _ _ (l₁.decidableSuffix l₂))
-      suffix_cons_iff.symm
-
-instance decidableInfix [DecidableEq α] : ∀ l₁ l₂ : List α, Decidable (l₁ <:+: l₂)
```
Comment (eric-wieser) [inline Mathlib/Data/List/Infix.lean:188]: Why was this deleted?
